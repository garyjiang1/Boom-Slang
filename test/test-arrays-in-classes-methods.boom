class Family:
	optional:
		string[][] relation = default string[4][2]

	def updateFathersName(string newName):
		self.relation[0][0] = newName

Family doeFamily = Family([["Typo","Stephanie","Jim","Kate"], ["father","mother","son","daughter"]])
doeFamily.updateFathersName("John")
int i = 0
loop i += 1 while i < 4:
	println(doeFamily.relation[0][i] + " is " + doeFamily.relation[1][i])
