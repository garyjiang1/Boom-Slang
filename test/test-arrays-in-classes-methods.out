John is father
Stephanie is mother
Jim is son
Kate is daughter
